drivers/hmis_external_apis/app/assets/javascripts/external_forms.js

```diff
@@ -111,29 +111,59 @@ $(function () {
   });
 });
 
-window.addDependentGroup = function (inputName, condValue, targetSelector) {
-  var target = $(targetSelector);
-  var show = function () {
+// conditions: [{input_name: 'name', input_value: 'value'}, ...]
+// targetSelector: selector for the item that is conditionally shown
+// enableBehavior: 'ANY' or 'ALL' conditions must be met to show the target selector
+window.addDependentGroup = function (conditions, targetSelector, enableBehavior = 'ANY') {
+  const target = $(targetSelector); // the item with enable_when on it
+  const show = function () {
     target.addClass('visible');
     target.attr('aria-hidden', "false");
     target.find('input, select, textarea').prop('disabled', false);
   }
-  var hide = function () {
+  const hide = function () {
     target.removeClass('visible');
     target.attr('aria-hidden', "true");
     target.find('input, select, textarea').prop('disabled', true);
   }
 
-  $('[name="' + inputName + '"]').on('change', function () {
-    var el = $(this)
-    var value = el.val();
-    if (el.prop('type') === 'checkbox') {
-      if (value === condValue) {
-        el.is(':checked') ? show() : hide();
+  // When *any* dependent item changes, this function will check all the conditions, and show/hide the target item accordingly.
+  const onDependentItemChanged = function() {
+    const evaluations = conditions.map(function ({ input_name, input_value }) {
+      const el = $('[name="' + input_name + '"]')
+      const input_type = el.prop('type');
+
+      // If the dependent item is a radio button item, we need to look at all the radio buttons with the same name, and find the one that is checked.
+      if (input_type === 'radio') {
+        const checked_val = $('[name="' + input_name + '"]:checked').val()
+        return checked_val === input_value;
+      }
+
+      const value = el.val()
+      if (input_type === 'checkbox') {
+        if (value === input_value) {
+          return el.is(':checked')
+        }
+      } else {
+        return value === input_value
       }
+      return false
+    });
+
+    const meetsCondition = enableBehavior === 'ALL' ? evaluations.every(Boolean) : evaluations.some(Boolean)
+    if (meetsCondition) {
+      show();
     } else {
-      value === condValue ? show() : hide();
+      hide();
     }
+  }
+
+  // add change listener to all dependent fields
+  const dependentItemSelectors = conditions.map(c => `[name="${c.input_name}"]`)
+  dependentItemSelectors.forEach(function (name) {
+    $(name).on('change', onDependentItemChanged);
   });
```

Comment:
not sure about compatibility with fat arrow and template literal. Maybe safer to avoid?
```suggestion
conditions.forEach(function (c) {
  const nameSelector = '[name="' + c.input_name + '"]';
  $(nameSelector).on('change', onDependentItemChanged);
});
```

Reply:
yep, thanks!